### PR TITLE
PHYDF-12 SSO OAUTH2

### PR DESCRIPTION
**Pull request by sendr**
[PHYDF-19](
https://youtrack.raccoongang.com/issue/PHYDF-19) - `SSO Oath2`
 - SSO Oauth2 RaccooGangCustomOAuth2
 - disabled the register and login pages (autoredirect to the provider)
 - added LOGOUT_REDIRECT_URL
 - wrapped pages like main_page, courses, course_about etc into the `@login_required`

### CODE REVIEW

File: common/djangoapps/student/middleware.py [hunk 1]
```diff
@@ -2,10 +2,13 @@
 Middleware that checks user standing for the purpose of keeping users with
 disabled accounts from accessing the site.
 """
-from django.http import HttpResponseForbidden
+from django.http import HttpResponseForbidden, HttpResponseRedirect
```
Comment (cmltaWt0) [inline common/djangoapps/student/middleware.py]: Unused import.

File: common/djangoapps/student/middleware.py [hunk 2]
```diff
@@ -2,10 +2,13 @@
...
 from django.utils.translation import ugettext as _
 from django.conf import settings
 from student.models import UserStanding
+from third_party_auth.provider import Registry
```
Comment (cmltaWt0) [inline common/djangoapps/student/middleware.py]: Unused import.

File: common/djangoapps/student/middleware.py [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
+from third_party_auth import pipeline
+from student.helpers import get_next_url_for_login_page, auth_pipeline_urls
```
Comment (cmltaWt0) [inline common/djangoapps/student/middleware.py]: Same as above.

File: common/djangoapps/student/middleware.py [hunk 4]
```diff
@@ -2,10 +2,13 @@
 Middleware that checks user standing for the purpose of keeping users with
 disabled accounts from accessing the site.
 """
-from django.http import HttpResponseForbidden
+from django.http import HttpResponseForbidden, HttpResponseRedirect
 from django.utils.translation import ugettext as _
 from django.conf import settings
 from student.models import UserStanding
+from third_party_auth.provider import Registry
+from third_party_auth import pipeline
```
Comment (cmltaWt0) [inline common/djangoapps/student/middleware.py]: Same as above.

File: common/djangoapps/student/views.py [hunk 5]
```diff
@@ -434,11 +434,18 @@ def _cert_info(user, course_overview, cert_status, course_mode):  # pylint: disa
 @ensure_csrf_cookie
 def signin_user(request):
     """Deprecated. To be replaced by :class:`student_account.views.login_and_registration_form`."""
+    # Determine the URL to redirect to following login:
+    redirect_to = get_next_url_for_login_page(request)
+    if not request.user.is_authenticated():
+        try:
+            provider = [_ for _ in third_party_auth.provider.Registry.get_enabled_by_backend_name('custom-oauth2')][0]
```
Comment (cmltaWt0) [inline common/djangoapps/student/views.py]: What if there are no enabled providers?

Comment (sendr) [reply common/djangoapps/student/views.py]: pass

Comment (sendr) [reply common/djangoapps/student/views.py]: we will see login/register page

Comment (cmltaWt0) [reply common/djangoapps/student/views.py]: Ups. I see :)

**Pull request merged**